src/tools/ci/test_extensions

```diff
@@ -0,0 +1,272 @@
+)
+
+# add postgis
+extensions.append(
+    {
+        'name': 'postgis',
+        'download': [
+            git_clone_command + ['https://github.com/postgis/postgis.git'],
+        ],
+        'build': [
+            ['./autogen.sh'],
+            ['./configure', '--without-raster'],
+            ['make'],
+        ],
+        'test': [
+            ['make', 'check'],
+        ],
+    }
+)
+
+# add pg_partman
+extensions.append(
+    {
+        'name': 'pg_partman',
+        'shared_preload_name': 'pg_partman_bgw',
+        'download': [
+            git_clone_command + ['https://github.com/pgpartman/pg_partman.git'],
+        ],
+        'build': [
+            ['make'],
+            ['sudo', 'make', 'install'],
+        ],
+        'test': [
+            # there is no test command,
+            # see bottom of https://github.com/pgpartman/pg_partman
+        ],
+    }
+)
+
+# add pgbouncer
+# since autoconf build is not working too, commented out for now
+# extensions.append(
+#     {
+#         'name': 'pgbouncer',
+#         'download': [
+#             git_clone_command + ['https://github.com/pgbouncer/pgbouncer.git'],
+#         ],
+#         'build': [
+#             ['git', 'submodule', 'init'],
+#             ['git', 'submodule', 'update'],
+#             ['./autogen.sh'],
+#             ['./configure', '--prefix=/usr/local'],
+#             ['make'],
+#             ['sudo', 'make', 'install'],
+#         ],
+#         'test': [
+#             ['make', '-C', './test', 'all'],
+#             ['make', '-C', './test', 'check'],
+#         ],
+#     }
+# )
+
+# filter elements of only-run-ext list
+if only_run_ext:
+    # check if all elements of only-list exists
+    ext_names = [elem['name'] for elem in extensions]
+    if not set(only_run_ext).issubset(ext_names):
+        sys.exit('There are unsupported extensions. Supported extensions are: '
+                 + ' '.join(ext_names))
+
+    extensions = [ext for ext in extensions if ext['name'] in only_run_ext]
+
+
+def start_postgres_server(logfile_prefix=''):
+    shared_preload_libraries = ''
+
+    # get shared_preload_libraries
+    for extension in extensions:
+        if 'shared_preload_name' in extension:
+            shared_preload_libraries = shared_preload_libraries + extension['shared_preload_name'] + ', '
+    # removing last ', '
+    shared_preload_libraries = shared_preload_libraries.rstrip(', ')
+
+    pgsql_data_path = '/usr/local/pgsql/data'
+    logfile_name = '{}-logfile'.format(logfile_prefix).strip('-')
+    postgres_start_commands = [
+        ['/usr/local/bin/pg_ctl', 'stop', '-D', pgsql_data_path],
+        ['sudo', 'rm', '-rf', pgsql_data_path],
+        ['sudo', 'mkdir', '-p', pgsql_data_path],
+        ['sudo', 'chown', getpass.getuser(), pgsql_data_path],
+        ['/usr/local/bin/initdb', '-D', pgsql_data_path],
+        ['sed', '-i',
+            's/#shared_preload_libraries = \'\'/shared_preload_libraries = \'' + shared_preload_libraries + '\'/g',
```

Comment:
I think there's a better solution to that: Create a file with the necessary additional configuration, and pass the file via make check TEMP_CONFIG=path-to-file